src/core/lwip/tcp_out.c

```diff
@@ -1885,6 +1885,44 @@ void tcp_rexmit_rto(struct tcp_pcb *pcb)
     if (pcb->unacked == NULL) {
         return;
     }
+    /*  Before moving all unacked to unsent, we need to maintain correct order of segments.
+        For exmaple, in unsent we might find a retransmitted segments:
+        unacked: 1->2->3
+        unsent: 4->5->6
+
+        After fast retransmission of seg 1,2:
+        unacked: 3
+        unsent: 1->2->4->5->6
+
+        After RTO without handling correct order:
+        unacked: ----
+        unsent: 3->1->2->4->5->6
+
+        Therefore - we first move back retransmitted segments to unacked:
+        unacked: 1->2->3
+        unsent: 4->5->6
+        And then move all unacked to unsent:
+        unacked: ----
+        unsent: 1->2->3->4->5->6
```

Comment:
No need to keep a long comment. It's enough to mention that "fast retransmission can insert out of order segment to the unsent queue".

Also fix the commit message : after a single fast retransmission the context will be:
unacked: 2->3
unsent:: 1->4->5->6
There is no straightforward way to get 1->2->4->5->6

Reply:
Done.